src/prototype_room_external.js

```diff
@@ -261,27 +261,13 @@ Room.prototype.checkAndSpawnReserver = function() {
       return;
     }
   }
-
-  let reserverSpawn = {
-    role: 'reserver',
-    level: 2,
-    routing: {
-      targetRoom: this.name,
-      targetId: this.controller.id,
-      reached: false,
-      routePos: 0,
-      pathPos: 0
-    }
-  };
   // TODO move the creep check from the reserver to here and spawn only sourcer (or one part reserver) when controller.level < 4
-  let energyNeeded = 1300;
+  let energyNeeded = baseRoom.getEnergyCapacityAvailable > 1300 ? baseRoom.energyCapacityAvailable / 4 : 1300;
```

Comment:
Shouldn't it be `4 * baseRoom.getEnergyCapacityAvailable > 1300 ? ...`. Otherwise on 1301` it you would return `325`